src/train_model.py

```diff
@@ -598,11 +639,26 @@ def eval_net(network, test_set, test_loader, device, criterion, dataset, _run):
                     data["output"] == predicted).all(-1).long().sum().item()
                 loss_sum += loss.item()
                 num_batches += 1
-
+            else:
+                inputs, labels = data[0].to(device), data[1].to(device)
+                outputs = network(inputs)
+                _, predicted = torch.max(outputs.data, 1)
+                loss = criterion(outputs, labels)
+                total += labels.size(0)
+                if dataset != "simple_dataset":
+                    # For simple dataset, it's a regression task, so
+                    # accuracy is meaningless
+                    correct += (predicted == labels).sum().item()
+                loss_sum += loss.item()
+                num_batches += 1
+    # For simple_dataset, test_accuracy will be 0, and should be ignored
+    # It is kept in to give the function a consistent interface
     test_accuracy = correct / total
     test_avg_loss = loss_sum / num_batches
-    _run.log_scalar("test." + test_set + ".accuracy", test_accuracy)
-    _run.log_scalar("test." + test_set + ".loss", test_avg_loss)
+    if _run is not None:
```

Comment:
_run should never be None, right?

Reply:
Ah, sorry, I added that when I wanted to import eval_net and run it in a Jupyter notebook. Removed